rag-and-text2sql-maharishi/workflow.py

```
@@ -0,0 +1,200 @@
+    """
+    Accepts user input --> LLM selects none or some tools -->
+    tools are collected and run --> tools result are appended
+    --> final LLM reply based on all the above appended context
+    """
+
+    def __init__(self,
+        tools: List[BaseTool],
+        timeout: Optional[float] = 60.0*20,
+        disable_validation: bool = False,
+        verbose: bool = True,
+        llm: Optional[LLM] = Ollama(model="qwen2.5:0.5b", temperature=0, request_timeout=60.0*5),
```

Ruff (0.8.2)

Do not perform function call `Ollama` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)